public/js/spreadsheet.js

```diff
@@ -29,37 +51,32 @@ document.addEventListener( 'DOMContentLoaded', () => {
       } );
     };
 
-    const getTextAtCell = ( row, index ) => row.children[ index ].textContent;
+    const getTextAtCell = ( index, row ) => row.children[ index ].textContent;
 
-    headerCells.forEach( ( headerCell, index ) => {
-      headerCell.addEventListener( 'click', () => {
-        defaultSortButton.style.display = 'inline-block';
-        let prevDir = headerCell.dataset.sortDir;
-        removeSortingAttributes();
-        headerCell.classList.add( 'spreadsheet-sorted-by' );
+    const sortRows = (headerCell, index) => () => {
+      let prevDir = headerCell.dataset.sortDir;
+      removeSortingAttributes();
+      headerCell.classList.add( 'spreadsheet-sorted-by' );
 
-        let sortedRows = rows.sort(
-          ( a, b ) => getTextAtCell( a, index ).localeCompare( getTextAtCell( b, index ) )
-        );
-        let newDir;
-        if ( prevDir ) {
-          newDir = prevDir === 'asc' ? 'desc' : 'asc';
-        } else {
-          newDir = 'asc';
+      rows = rows.sort( ( rowA, rowB ) => {
+        const textCompare = getTextAtCell( index, rowA ).localeCompare( getTextAtCell( index, rowB ) );
+        if ( textCompare !== 0 ) {
+          return textCompare;
         }
-        headerCell.dataset.sortDir = newDir;
-        if ( newDir === 'desc' ) {
-          sortedRows = sortedRows.reverse();
+        // If text is the same preserve the original sort order.
+        return rowA.dataset.order - rowB.dataset.order;
         }
-        table.append( ...sortedRows );
-      } );
-    } );
+      ); const newDir = prevDir !== 'asc' ? 'asc' : 'desc';
```

Comment:
There seems to be a indentation issue in line 68 (ESLint complained), and I guess this line could probably be in a new line. This is totally a nit, feel free to ignore.